lab_2_retrieval_w_bm25/main.py

```diff
@@ -78,6 +134,28 @@ def calculate_idf(vocab: list[str], documents: list[list[str]]) -> dict[str, flo
     In case of corrupt input arguments, None is returned.
     """
 
+    if not vocab or not isinstance(vocab, list) \
+            or not documents or not isinstance(documents, list) \
+            or not all(isinstance(tok_doc, list) for tok_doc in documents) \
+            or not all(tok_doc for tok_doc in documents):
+        return None
+
+    idf_vocab = {}
+    for un_word in vocab:
+        if not un_word or not isinstance(un_word, str):
+            return None
+        count = 0
+        for doc in documents:
+            for i in doc:
+                if not i or not isinstance(i, str):
+                    return None
+            if un_word in doc:
+                count += 1
+            else:
+                continue
```

Comment:
whyyyy?

Reply:
to calculate how many documents contain this word, i don't know how to do it any other way :'(

Reply:
it is ugly, find more elegant way